packages/core-data/src/types/README.md

```diff
@@ -192,3 +192,32 @@ A string that the server renders which often involves modifications from the raw
 
 For example, block HTML with the comment delimiters exists in `post_content` but those comments are stripped out when rendering to a page view. Similarly, plugins might modify content or replace shortcodes.
 
+## Extending
+
+You can extend the entity record definitions using [TypeScript's declaration merging](https://www.typescriptlang.org/docs/handbook/declaration-merging.html).
+
+For example, if you're building a plugin that displays a number of views of each comment, you can add a new `numberOfViews` field to the `Comment` type like this:
+
+```ts
+// In core-data
+export interface ExtensibleComment< C extends Context > {
+	id: number;
+	// ...
+}
+
+export type Comment< C extends Context > = OmitNevers<
+	ExtensibleComment< C >
+>;
+
+// In the plugin
+import { ExtensibleComment, Comment } from '@wordpress/core-data';
+interface ExtensibleComment < C extends Context > {
+	numberOfViews: number;
+}
+
+const c : Comment = ...
+
+// c.numberOfViews is a number
```

Comment:
can we show that `id` and other properties are still here? this shows we can add `numberOfViews` but leaves me wondering if it will erase everything else.